### ISSUE TITLE
Data type must be one of [float32, float64, int16, int32], not float16

### ISSUE THREAD

**Issue by ckinpdx** [wildminder/ComfyUI-VibeVoice#6]
I placed this node in the KJ WANS2V wf in place of input audio and got the subject line error at the melbandroformer node. I made the below change and it seems to be working correctly now.

        output_waveform = outputs.speech_outputs[0]
        if output_waveform.ndim == 1: output_waveform = output_waveform.unsqueeze(0)
        if output_waveform.ndim == 2: output_waveform = output_waveform.unsqueeze(0)

        output_waveform = output_waveform.to(torch.float32)
        return ({"waveform": output_waveform.detach().cpu(), "sample_rate": 24000},)

**Comment by wildminder** [comment]
Does this error still persist with the latest updates?
